src/Core/SettingsQuirks.cpp

```diff
@@ -14,19 +15,22 @@
 ///   [2]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=0c54a6a44bf3
 bool nestedEpollWorks(Poco::Logger * log)
 {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)) && (LINUX_VERSION_CODE < KERNEL_VERSION(5, 6, 13))
-        /// the check is correct since there will be no more 5.5.x releases.
+    if (Poco::Environment::os() != POCO_OS_LINUX)
+        return true;
+
+    DB::VersionNumber linux_version = DB::VersionNumber::fromString(
+        Poco::Environment::osVersion(), /* strict= */ false);
+
+    /// the check is correct since there will be no more 5.5.x releases.
+    if (linux_version >= DB::VersionNumber{5, 5, 0} && linux_version < DB::VersionNumber{5, 6, 13})
+    {
```

Comment:
Let's make a plan on how soon we will remove this code completely.
Maybe 2022-01-01?

Reply:
Well, if something that does not in control of clickhouse has an issue it should not be removed until it will past enough time (1-2 years).
Consider linux kernel, it has lot's of quirks for hardware.

And also note, that this is pretty nasty issue (I spend quite a lot of time to understand that the issue is in the kernel not in the async_socket_for_remote or/and use_hedged_requests), so that said that to me, this quirk brings more benefits than break "code clearness".